NETCORE/src/Shared/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -149,6 +153,15 @@ public void Configure(TelemetryConfiguration configuration)
                     // So this approach of adding all modules to DI, but selectively
                     // disable those modules which user has disabled is chosen.
 
+                    if (module is DiagnosticsTelemetryModule)
+                    {
+                        if (!this.applicationInsightsServiceOptions.EnableDiagnosticsTelemetryModule)
+                        {
+                            DisposeIfDisposable(module);
```

Comment:
The heartbeatmanager is extraced and enabled independent of this, which likely means they operate on disposed heartbeatmanager, when one does EnableDiagnosticsTelemetryModule=false.

Reply:
solved